README.md

```diff
@@ -40,6 +45,9 @@ Contacts.getAll((err, contacts) => {
   }
 })
 ```
+**Note** `getAll` is a database intensive process, and can take a long time to complete depending on the size of the contacts list. Because Android is particularly slow, it is recommended you use `get(options, …)` to try to optimize your query, to only fetch contacts that are relevant to your use case. See example above.
```

Comment:
I took out the note about caching results for future use, because this doesn't seem like a practical solution.  On both Android and iOS on contemporary OSes (iOS 9,10 and Android M, N), you must ask for permission first before being able to query contacts.  Thus, pre-caching of contacts doesn't really help, because the user will be prompted for permissions at that time.